ehrapy/anndata/anndata_ext.py

```diff
@@ -176,17 +178,30 @@ def move_to_obs(adata: AnnData, to_obs: list[str] | str, copy: bool = False) ->
         raise ObsMoveError(
             "Cannot move encoded columns from X to obs. Either undo encoding or remove them from the list!"
         )
-    indices = adata.var_names.isin(to_obs)
-    df = adata[:, indices].to_df()
-    adata._inplace_subset_var(~indices)
-    adata.obs = adata.obs.join(df)
-    updated_num_uns, updated_non_num_uns, num_var = _update_uns(adata, to_obs)
-    # cast numerical values from object
-    adata.obs[num_var] = adata.obs[num_var].apply(pd.to_numeric, errors="ignore", downcast="float")
-    # cast non numerical values from object to either bool (if possible) or category
-    adata.obs = _cast_obs_columns(adata.obs)
-    adata.uns["numerical_columns"] = updated_num_uns
-    adata.uns["non_numerical_columns"] = updated_non_num_uns
+
+    if copy_obs:
+        indices = adata.var_names.isin(to_obs)
+        df = adata[:, indices].to_df()
+        adata.obs = adata.obs.join(df)
+        updated_num_uns, updated_non_num_uns, num_var = _update_uns(adata, to_obs)
```

Comment:
By copying columns from X to obs, the columns still remain in X. So they should not got deleted from uns numerical/non numerical (which is what `update_uns` is basically doing). 
This also requires changes to `move_to_x`, which should check whether the columns are already present in X (in that case simply delete those columns from obs).

Reply:
So I just have to delete the lines below ```update_uns```?